src/Hedgehog/Extras/Test/File.hs

```diff
@@ -336,3 +337,15 @@ appendFileTimeDelta filePath offsetTime = GHC.withFrozenCallStack $ do
   baseTime <- H.noteShowIO DTC.getCurrentTime
   let delay = DTC.diffUTCTime baseTime offsetTime
   appendFile filePath $ show @DTC.NominalDiffTime delay <> "\n"
+
+-- | Asserts that the given directory exists.
+assertDirectoryExists :: (MonadTest m, MonadIO m, HasCallStack) => FilePath -> m ()
+assertDirectoryExists dir = GHC.withFrozenCallStack $ do
+  exists <- H.evalIO $ IO.doesDirectoryExist dir
+  unless exists $ H.failWithCustom GHC.callStack Nothing (dir <> " has not been successfully created.")
+
+-- | Asserts that the given directory is missing.
+assertDirectoryMissing :: (MonadTest m, MonadIO m, HasCallStack) => FilePath -> m ()
+assertDirectoryMissing dir = GHC.withFrozenCallStack $ do
+  exists <- H.evalIO $ IO.doesDirectoryExist dir
+  when exists $ H.failWithCustom GHC.callStack Nothing (dir <> " should not have been created.")
```

Comment:
This function has nothing to do with directory creation, so I suggest changing error message a bit:
```suggestion
  when exists $ H.failWithCustom GHC.callStack Nothing ("Directory " <> dir <> " does not exist on the file system.")
```